/***************************************************************************
**                                                                        **
**  FlySight Viewer                                                       **
**  Copyright 2018 Michael Cooper, Klaus Rheinwald                        **
**                                                                        **
**  This program is free software: you can redistribute it and/or modify  **
**  it under the terms of the GNU General Public License as published by  **
**  the Free Software Foundation, either version 3 of the License, or     **
**  (at your option) any later version.                                   **
**                                                                        **
**  This program is distributed in the hope that it will be useful,       **
**  but WITHOUT ANY WARRANTY; without even the implied warranty of        **
**  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         **
**  GNU General Public License for more details.                          **
**                                                                        **
**  You should have received a copy of the GNU General Public License     **
**  along with this program.  If not, see <http://www.gnu.org/licenses/>. **
**                                                                        **
****************************************************************************
**  Contact: Michael Cooper                                               **
**  Website: http://flysight.ca/                                          **
****************************************************************************/

#ifndef PPCFORM_H
#define PPCFORM_H

#include <QWidget>

namespace Ui {
class PPCForm;
}

class MainWindow;

class PPCForm : public QWidget {
    Q_OBJECT

public:
    explicit PPCForm(QWidget* parent = 0);
    ~PPCForm();

    virtual QSize sizeHint() const;

    void setMainWindow(MainWindow* mainWindow);

protected:
    virtual void keyPressEvent(QKeyEvent*);

private:
    Ui::PPCForm* ui;
    MainWindow* mMainWindow;

public slots:
    void updateView();

private slots:
    void onFAIButtonClicked();
    void onUpButtonClicked();
    void onDownButtonClicked();
    void onApplyButtonClicked();
    void onEndButtonClicked();

    void onModeChanged();

    void onActualButtonClicked();
    void onOptimalButtonClicked();
    void onOptimizeButtonClicked();

    void onPpcButtonClicked();
};

#endif // PPCFORM_H
